### ISSUE TITLE
socket.error: [Errno 10054] An existing connection was forcibly closed by the remote host

### ISSUE THREAD

**Issue by jtpereyda** [jtpereyda/boofuzz#85]
One user has reported and I've personally seen errors like this before:

```
Traceback (most recent call last):
  File "Z:\ftp\ftp.py", line 119, in <module>
    sess.fuzz()
  File "C:\Python27\lib\site-packages\boofuzz\sessions.py", line 414, in fuzz
    self._fuzz_current_case(*fuzz_args)
  File "C:\Python27\lib\site-packages\boofuzz\sessions.py", line 868, in _fuzz_current_case
    self.transmit(target, self.fuzz_node, edge)
  File "C:\Python27\lib\site-packages\boofuzz\sessions.py", line 739, in transmit
    self.last_recv = self.targets[0].recv(10000)
  File "C:\Python27\lib\site-packages\boofuzz\sessions.py", line 110, in recv
    data = self._target_connection.recv(max_bytes=max_bytes)
  File "C:\Python27\lib\site-packages\boofuzz\socket_connection.py", line 143, in recv
    data = self._sock.recv(max_bytes)
socket.error: [Errno 10054] An existing connection was forcibly closed by the remote host
```

This seems to indicate a crash of the unit under test, which boofuzz should report as an error instead of crashing itself.

I don't know if there are times when this is valid behavior; perhaps in some protocols.

If anyone has any further details or ideas, please post.

**Comment by n30m1nd** [comment]
I would say this could be normal behaviour as long as you can replicate the behaviour/crash or start from the point it crashed (which I suppose are already included features on boofuzz). Aren't these?

**Comment by jtpereyda** [comment]
@n30m1nd When I've encountered this error, the case is usually reproducible based on the log output leading up to the error.

**Comment by jtpereyda** [comment]
Fixed between Pull Requests #86, #90, and #93. Big thanks to @marpie for serious footwork and patience with this debugging effort.
